### PR TITLE
feat: resolver on hotdashboard

### PR DESCRIPTION
**Pull request by KngZhi**
hot dashboard resolver on stats page.

ref: https://github.com/kodadot/nft-gallery/issues/3114

image

### CODE REVIEW

**Comment by vikiival** [comment]
Screenshot 2022-06-11 at 15 24 46
this does not look safe

**Comment by vikiival** [comment]
Plase also add screenshot from the playground

**Comment by KngZhi** [comment]
> Plase also add screenshot from the playground

Okay, updated. :)

**Review by vikiival — CHANGES_REQUESTED** [review]

File: src/server-extension/query/nft.ts [hunk 1]
```diff
@@ -21,4 +21,19 @@ FROM nft_entity ne
...
+    e.meta,
+    ce.name as collection_name,
+    ce.id as collection_id
+FROM
+    event e
+    LEFT JOIN nft_entity ne ON e.nft_id = ne.id
+    LEFT JOIN collection_entity ce ON ne.collection_id = ce.id
+WHERE
+    e.interaction = 'BUY'
+    AND e."timestamp" >= NOW() - INTERVAL '${dateRange}'
```
[the remark this answers is not in the document]
Comment (KngZhi) [reply src/server-extension/query/nft.ts:37]: for `$1` or `'$1'`, both would raise errors.

image

File: src/server-extension/query/nft.ts [hunk 2]
```diff
@@ -21,4 +21,19 @@ FROM nft_entity ne
 where
     e.interaction = 'BUY'
     and e.timestamp >= NOW() - INTERVAL '7 DAY'
-ORDER BY e.timestamp desc`
\ No newline at end of file
+ORDER BY e.timestamp desc`
```
Comment (vikiival) [inline src/server-extension/query/nft.ts:24]:
```suggestion
ORDER BY e.timestamp DESC`
```

**Review by vikiival — CHANGES_REQUESTED** [review]
Smol stuff

**Comment by vikiival** [comment]
Screenshot 2022-10-18 at 14 48 24

**Review by vikiival — APPROVED** [review]

**Pull request merged**